CMakeLists.txt

```diff
@@ -472,239 +472,14 @@ add_executable(smb
-    test/bwe/BandwidthUtilsTest.cpp
-    test/bwe/EstimatorTestEasy.cpp
-    test/bwe/EstimatorReRun.cpp
-    test/crypto/AESTest.cpp
-    test/utils/Base64Test.cpp
-    test/crypto/AesIvGeneratorTest.cpp
-    test/transport/RecordingTransportTest.cpp
-    test/transport/recp/RecStartStopEventBuilderTest.cpp
-    test/transport/recp/RecStreamAddedEventBuilderTest.cpp
-    test/bridge/UnackedPacketsTrackerTest.cpp
-    test/bridge/VideoForwarderRtxReceiveJobTest.cpp
-    test/memory/PriorityQueueTest.cpp
-    test/memory/BacklogTest.cpp
-    test/memory/StackMapTest.cpp
-    test/bridge/ActiveMediaListTestLevels.h
-    test/bridge/MixerTest.cpp
-    test/bridge/VideoNackReceiveJobTest.cpp
-    test/utils/LogSpamTest.cpp
-    test/utils/FunctionTest.cpp
-    test/transport/JitterTest.cpp)
-
-
-set(TEST_FILES2
-    test/integration/RealTimeTest.cpp
-    test/integration/RealTimeTest.h
-    test/integration/BarbellTest.cpp
-    test/integration/BarbellTest.h
-    test/integration/ConfIntegrationTest.cpp
-    test/integration/IntegrationLegApiTest.cpp
-    test/integration/IceTransportTest.cpp
-    test/integration/SrtpIntegrationTest.cpp
-    test/integration/IntegrationCallTypes.cpp
-    test/integration/IntegrationAudioTest.cpp
-    test/integration/FFTtest.cpp
-)
-
-set(LOAD_TEST_FILES
-    test/integration/IntegrationTest.cpp
-    test/integration/IntegrationTest.h
-    test/integration/RealTimeTest.cpp
-    test/integration/RealTimeTest.h
-    test/integration/LoadTestConfig.h
-)
-
-
 set(CMAKE_TEST_DIRECTORY "${CMAKE_BINARY_DIR}/test")
 set(CMAKE_TEST_INCLUDE "${CMAKE_TEST_DIRECTORY}/include")
 configure_file(test/TestConfig.h.in ${CMAKE_TEST_INCLUDE}/TestConfig.h)
 
-add_library(testlib STATIC ${TEST_LIB_FILES})
-target_include_directories(testlib PRIVATE ${CMAKE_TEST_INCLUDE} ${gtest_SOURCE_DIR}/include "test/include")
-target_link_libraries(testlib PUBLIC smblib ${THIRD_PARTY_LIBS})
-
-add_executable(UnitTest
-    ${TEST_FILES} test/gtest_main.cpp)
-
-target_include_directories(UnitTest PRIVATE ${CMAKE_TEST_INCLUDE} "test/include")
-
-# Copy test Data
-file(COPY "${CMAKE_SOURCE_DIR}/test/resources" DESTINATION "${CMAKE_TEST_DIRECTORY}")
-
-target_link_libraries(UnitTest testlib gtest gmock)
-
-add_executable(UnitTest2
-    ${TEST_FILES2} test/gtest_main2.cpp)
-
-target_include_directories(UnitTest2 PRIVATE ${CMAKE_TEST_INCLUDE} "test/include")
-
-# Copy test Data
-file(COPY "${CMAKE_SOURCE_DIR}/test/resources" DESTINATION "${CMAKE_TEST_DIRECTORY}")
-
-target_link_libraries(UnitTest2 testlib gtest gmock)
-
-
-add_test(AllUnitTests UnitTest)
-add_test(AllUnitTests UnitTest2)
-
-add_executable(LoadTest ${LOAD_TEST_FILES}
-    test/load_test_main.cpp)
-
-target_link_libraries(LoadTest testlib gtest gmock)
+add_subdirectory(test)
+set_target_properties(UnitTest PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR} ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR} RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
```

Comment:
This is to make Test executables end up in build top folder. 
However, maybe we should consider allowing the build folder to be structured as cmake default:
build
  bin
  _deps
  libs